cmd/otel-allocator/main.go

```diff
@@ -59,6 +60,9 @@ func main() {
 		setupLog.Error(err, "Unable to initialize allocation strategy")
 		os.Exit(1)
 	}
+
+	filterStrategy := cfg.GetTargetsFilterStrategy()
```

Comment:
should do something similar to line 58 `allocation.New`? This will make it easier to generalize and add more strategies later down the line